build/test-threadSafety.ps1

```diff
@@ -0,0 +1,35 @@
+param(
+  [Parameter(Mandatory=$true)][string]$coyoteVersion="",
+  [Parameter(Mandatory=$true)][string]$targetFramework,
+  [Parameter(Mandatory=$true)][string]$methodNames,
+)
+
+$rootDirectory = Split-Path $PSScriptRoot -Parent
+
+Write-Host "Install Coyote CLI."
+dotnet tool install --global Microsoft.Coyote.CLI
+
+Write-Host "Build OpenTelemetry.Tests project."
+dotnet build $rootDirectory\test\OpenTelemetry.Tests\OpenTelemetry.Tests.csproj
+
+$artifactsPath = Join-Path $rootDirectory "test\OpenTelemetry.Tests\bin\Debug\$targetFramework"
+
+Write-Host "Generate Coyote rewriting options JSON file."
```

Comment:
Generate Json file for Coyote to pass in when doing rewriting the binaries, i.e.: adding scheduling points to explore different interleaving sequences between different threads.

Json file looks like this:
```json
{
    "AssembliesPath": ".\\bin\\Debug\\net7.0",
     "Assemblies": [
       "OpenTelemetry.Api.dll",
       "OpenTelemetry.Api.ProviderBuilderExtensions.dll",
       "OpenTelemetry.dll",
       "OpenTelemetry.Exporter.InMemory.dll",
       "OpenTelemetry.Tests.dll"
   ]
}
```